### ISSUE TITLE
Remove rack version dependency

### ISSUE THREAD

**Issue by klyonrad** [arangamani/jenkins_api_client#215]
Rails 5 depends on `actionpack` which depends on `rack` above Version 2.

Is there any reason why this Gem depends on `rack, '~> 1.0'` since Version 1.4.4? I could not find an explanation.

```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rails (~> 5.0) was resolved to 5.0.0, which depends on
      actionmailer (= 5.0.0) was resolved to 5.0.0, which depends on
        actionpack (= 5.0.0) was resolved to 5.0.0, which depends on
          rack (~> 2.0)

    jenkins_api_client (~> 1.4.2) was resolved to 1.4.4, which depends on
      rack (~> 1.0)

    rails (~> 5.0) was resolved to 5.0.0, which depends on
      actionmailer (= 5.0.0) was resolved to 5.0.0, which depends on
        actionpack (= 5.0.0) was resolved to 5.0.0, which depends on
          rack-test (~> 0.6.3) was resolved to 0.6.3, which depends on
            rack (>= 1.0)
```

**Comment by chilicheech** [comment]
This is affecting us as well. We use `chef` and it now requires `rack ~> 2`. Will `jenkins_api_client` not work with `rack ~> 2`?

**Comment by stjohnjohnson** [comment]
@arangamani do you know if this is used?  It's blocking some of our upgrades.

**Comment by stjohnjohnson** [comment]
@arangamani any updates?

**Comment by arangamani** [comment]
It is only used as a dev dependency. Fixed and released 1.4.5.
